Gradio 5 release blog post (#2398)
huggingface/blog: gradio-5.md

Our goal with Gradio 5 was to listen to and address the most common pain points that we’ve heard from Gradio developers about building production-ready Gradio apps. For example, we’ve heard some developers tell us:

*   “Gradio apps load too slowly” → Gradio 5 ships with major performance improvements, including the ability to serve Gradio apps via server-side rendering (SSR) which loads Gradio apps almost instantaneously in the browser. _No more loading spinner_! We also no longer render components in the DOM that are not visible, which should significantly reduce the memory footprint of larger Gradio apps 🏎️💨

<video width="600" controls>
  <source src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/gradio-5/gradio-4-vs-5-load.mp4">
</video>

*   "This Gradio app looks old-school" → Many of the core Gradio components, including Buttons, Tabs, Sliders, as well as the high-level chatbot interface, have been refreshed with a more modern design in Gradio 5. We’re also releasing a new set of built-in themes, to let you easily create fresh-looking Gradio apps 🎨

> *   “I can’t build realtime apps in Gradio” → We have changed how streaming works internally: we now use websockets and send data via base64 to minimize latency. We [support WebRTC via custom components](https://huggingface.co/spaces/freddyaboulton/webrtc), and have also added significantly more documentation and example demos that are focused on common streaming use cases, such as webcam-based object detection, video streaming, real-time speech transcription and generation, and [conversational chatbots](https://huggingface.co/spaces/gradio/omni-mini). 🎤

Reviewer: ```suggestion
*   “I can’t build realtime apps in Gradio” → We have changed how streaming works internally: we now use websockets and send data in base64 to minimize latency. We [support WebRTC via custom components](https://huggingface.co/spaces/freddyaboulton/webrtc), and have also added significantly more documentation and example demos that are focused on common streaming use cases, such as webcam-based object detection, video streaming, real-time speech transcription and generation, and [conversational chatbots](https://huggingface.co/spaces/gradio/omni-mini). 🎤
```

- base64 does not seem the most efficient format, maybe briefly explain how it used to be?
- The demo does not work for me right now (I get a spinner when I click Record)
-  I'm not sure about the implications of WebRTC via "custom components". What is the recommended starting path for streaming use cases? Perhaps we can explicitly link to it as a recommendation to "get started with streaming"

Author: I removed the link to the demos for now. We are in the process of publishing docs to our website today, when they are up, I'll open a PR to add the links here

*   "LLMs don't know Gradio" → Gradio 5 ships with an experimental AI Playground where you can use AI to generate or modify Gradio apps and preview the app right in your browser immediately: [https://www.gradio.app/playground](https://www.gradio.app/playground) 

<video width="600" controls>
  <source src="https://huggingface.co/datasets/huggingface/documentation-images/raw/main/blog/gradio-5/simple-playground.mp4">
</video>

> Gradio 5 fixes these issues while maintaining Gradio’s simple and intuitive developer-facing API. Since Gradio 5 intended to be a production-ready web framework for all kinds of machine learning applications, we've also made significant improvements around web security (including getting a 3rd-party audit of Gradio) -- more about that in an upcoming post!

Reviewer: ```suggestion
Gradio 5 provides all these features while maintaining Gradio’s simple and intuitive developer-facing API. Since Gradio 5 intended to be a production-ready web framework for all kinds of machine learning applications, we've also made significant improvements around web security (including getting a 3rd-party audit of Gradio) -- more about that in an upcoming post!
```

> ### Breaking changes

Reviewer: ```suggestion
## Breaking changes
```

Gradio apps that did not raise any deprecation warnings in Gradio 4.x should continue to work in Gradio 5, with a [small number of exceptions. See a list of breaking changes in Gradio 5 here](https://github.com/gradio-app/gradio/issues/9463). 

> ### What’s next for Gradio?

Reviewer: ```suggestion
## What’s next for Gradio?
```

Many of the changes we’ve made in Gradio 5 are designed to enable new functionality that we will be shipping in the coming weeks. Stay tuned for:

*   Multi-page Gradio apps
    
*   Navbars and sidebars
    
*   Support for running Gradio apps on mobile using PWA and potentially native app support
    

> *   More media components to support new modalities that emerging around images and videos

Reviewer: ```suggestion
*   More media components to support emerging modalities around images and videos
```

    
*   A richer DataFrame component with support for common spreadsheet-type operations

>     

Reviewer 2: We are also releasing another new functionality that allows 'src' to accept a 'ModelRegistry' callable. This callable can take a string and return a Gradio Interface app. 
So, should we add 'Constructing Gradio apps from a 3rd-party API model name automatically' to the list? 
I feel that including this information in the official release blog can be beneficial for our outreach regarding these integrations.
